DROOLS-7115 Drools 8 docs : Release Notes (#4668)
kiegroup/drools: drools-docs/src/modules/ROOT/pages/release-notes/index.adoc

:DROOLS:
include::../_artifacts/document-attributes.adoc[]

[[release.notes]]
:context: release-notes

[id='{context}']
= Release Notes

This chapter contains the release notes for the {PRODUCT} 8-series.

> For the release notes of the _previous releases_, you can make reference to the {PRODUCT} documentation of link:https://docs.drools.org/7.73.0.Final/drools-docs/html_single/index.html#droolsreleasenoteschapter[version 7]. 

Reviewer: ```suggestion
For the release notes of the _previous releases_, you can refer to the {PRODUCT} documentation of link:https://docs.drools.org/7.73.0.Final/drools-docs/html_single/index.html#droolsreleasenoteschapter[version 7]. 
```

== Drools 8-series release notes

> Drools 8-series is a natural evolution of the 7.x-stream, incorporating many features and lesson learned integrating with Kogito and many cloud-native use cases.

Reviewer: ```suggestion
Drools 8-series is a natural evolution of the 7.x-stream, incorporating many features and lessons learned integrating with Kogito and many cloud-native use cases.
```

=== Rule Unit

Rule Unit is a central paradigm in the Drools 8-series.

> You can make reference to the xref:migration-guide/index.adoc[Migration guide chapter] for more information about Rule Unit.

Reviewer: ```suggestion
You can refer the xref:migration-guide/index.adoc[Migration guide chapter] for more information about Rule Unit.
```

=== Minimum requirements update

JDK 11 is now the minimum Java version required to compile Drools and make use of Drools.

> Maven `3.8.4` is now the minimum Maven version required to build Drools from source, or using `kie-ci` APIs.

Reviewer 2: I think it should be `3.8.6`, please double check.

Author: It's 3.8.4 defined here:

https://github.com/kiegroup/drools/blob/6a7f986bd0a6f551c16c2b04939951ca29c38b19/build-parent/pom.xml#L80

Maybe you intend to update it there @mariofusco ?

Reviewer 2: Correct me if I'm wrong, but I think that Kogito requires maven 3.8.6 and if so I don't see any reason to have Drools and Kogito not aligned on this. So yes, I'd change also the version in the pom.

Author: https://issues.redhat.com/browse/DROOLS-7149
👉  https://github.com/kiegroup/drools/pull/4679

> === Mvel is deprecated
> 
> `drools-mvel` is now deprecated in favor of the executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. It doesn't affect how to write rules.

Reviewer 3: ```suggestion
=== `drools-mvel` is deprecated

`drools-mvel` is now deprecated in favor of the executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. You can still use MVEL syntax in your rules, so it doesn't affect how to write rules.
```

Author: @danielezonca before I merge Toshiya's suggestion, is that good for you pls?

Reviewer 4: LGTM 👍

> `drools-mvel` is now deprecated in favor of executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. It doesn't affect how to write rules.

Reviewer: ```suggestion
`drools-mvel` is now deprecated in favor of the executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. It doesn't affect how to write rules.
```

> `drools-mvel` is now deprecated in favor of executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. It doesn't affect how to write rules.

Author: do you have a better wording proposal, please?
/cc @mariofusco @tkobayas

> `drools-mvel` is now deprecated in favor of executable model. It also means `drools-engine` is recommended instead of deprecated `drools-engine-classic` that contains `drools-mvel`. It doesn't affect how to write rules.

Reviewer 3: See my suggestion below.

> === Security Manager deprecation

Reviewer: ```suggestion
=== Security manager deprecation
```

Author: Sorry but JDK refers to this in all caps. Keeping as-is.
ref https://openjdk.org/jeps/411

Author: Discussed with Kalyani, we will indeed keep as-is for the reasons above.

> The {PRODUCT} features related to the JDK Security Manager will not be further developed, as starting with JDK17, the Java Platform has deprecated the Security Manager for removal.

Reviewer: ```suggestion
The {PRODUCT} features related to the JDK security manager will not be further developed, as starting with JDK17, the Java platform has deprecated the security manager for removal.
```

Author: Sorry but JDK refers to this in all caps. Keeping as-is.
ref https://openjdk.org/jeps/411

Author: Discussed with Kalyani, we will indeed keep as-is for the reasons above.

> You can make reference to the xref:KIE/index.adoc#_securitymanager[Security Manager chapter] for more information.

Reviewer: ```suggestion
You can refer the xref:KIE/index.adoc#_securitymanager[Security Manager chapter] for more information.
```

=== KIE Marshallers notice

> The `KieMarshallers` are still present and working as expected on the release of the Drools 8-series; we are currently exploring however alternative solutions more idiomatic to cloud-native use cases.

Reviewer 5: ```suggestion
The `KieMarshallers` are still present and working as expected on the release of the Drools 8-series, however we are currently exploring alternative solutions more idiomatic to cloud-native use cases.
```

> The `KieMarshallers` are still present and working as expected on the release of the Drools 8-series, however we are currently exploring alternative solutions more idiomatic to cloud-native use cases.

Reviewer: Instead of _**idiomatic**_ we can use _**conversational**_ as it is more readable and easy to understand. WDYT?
```suggestion
The `KieMarshallers` are still present and working as expected on the release of the Drools 8-series, however, we are currently exploring alternative solutions more conversational to cloud-native use cases.
```

Author: Honestly I have no idea what "conversational" means here? 🤔 
Idiomatic instead in programming is used to refer to "best usage like taken from the manual/like seen in the course"

Author: Discussed with Kalyani, we will indeed keep as-is for the reasons above.

== Previous release notes

> For the release notes of the previous releases, you can make reference to the {PRODUCT} documentation of link:https://docs.drools.org/7.73.0.Final/drools-docs/html_single/index.html#droolsreleasenoteschapter[version 7]. 

Reviewer: ```suggestion
For the release notes of the previous releases, you can refer the {PRODUCT} documentation of link:https://docs.drools.org/7.73.0.Final/drools-docs/html_single/index.html#droolsreleasenoteschapter[version 7]. 
```
